TS6133 Error if using typescript with noUnusedLocals and typescript preprocessor

The functions declared in the output parser cause TS6133 errors when attempting to compile with typescript. This is because understandably, not all functions that are provided by builtins and generators are used in a user's grammar.

How to reproduce:

**grammar.ne**
```
@preprocessor typescript
@builtin "postprocessors.ne"

main -> foo
foo -> "foo"i
```

```
bash-3.2$ tsc -v
Version 2.6.1

bash-3.2$ nearleyc -v
2.11.0

bash-3.2$ nearleyc grammar.ne -o parser.ts

bash-3.2$ tsc --noUnusedLocals
parser.ts: error TS6133: 'id' is declared but never used.
parser.ts: error TS6133: 'nth' is declared but never used.
parser.ts: error TS6133: '$' is declared but never used.
```

--- a/builtin/postprocessors.ne
+++ b/builtin/postprocessors.ne
@@ -3,6 +3,8 @@
 # Postprocessor generator that lets you select the nth element of the list.
 # `id` is equivalent to nth(0).
 @{%
+// Bypasses TS6133. Allow declared but unused functions.
+// @ts-ignore
 function nth(n) {
     return function(d) {
         return d[n];
@@ -12,6 +14,8 @@ function nth(n) {
 
 # Postprocessor generator that lets you generate an object dynamically.
 @{%
+// Bypasses TS6133. Allow declared but unused functions.
+// @ts-ignore
 function $(o) {
     return function(d) {
         var ret = {};

--- a/lib/generate.js
+++ b/lib/generate.js
@@ -181,6 +181,8 @@
     generate.ts = generate.typescript = function (parser, exportName) {
         var output = "// Generated automatically by nearley, version " + parser.version + "\n";
         output +=  "// http://github.com/Hardmath123/nearley\n";
+        output +=  "// Bypasses TS6133. Allow declared but unused functions.\n";
+        output +=  "// @ts-ignore\n";
         output += "function id(d: any[]): any { return d[0]; }\n";
         output += parser.customTokens.map(function (token) { return "declare var " + token + ": any;\n" }).join("")
         output += parser.body.join('\n');

--- a/package.json
+++ b/package.json
@@ -55,6 +55,6 @@
     "microtime": "^2.1.2",
     "mocha": "^2.5.3",
     "moo": "^0.3.2",
-    "typescript": "^2.3.4"
+    "typescript": "^2.6.1"
   }
 }

